packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -3196,15 +3206,26 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     log.debug('handleAggregationChange', aggregation);
 
     this.startLoading(`Aggregating ${aggregation.operation}...`);
-    this.setState(({ aggregationSettings }) => ({
-      selectedAggregation: aggregation,
-      aggregationSettings: {
-        ...aggregationSettings,
-        aggregations: aggregationSettings.aggregations.map(a =>
-          a.operation === aggregation.operation ? aggregation : a
-        ),
-      },
-    }));
+    this.setState(({ aggregationSettings }) => {
+      let newAggregations = [...aggregationSettings.aggregations];
+      let aggregationIndex = newAggregations.findIndex(
+        a => a.operation === aggregation.operation
+      );
+      if (aggregationIndex === -1) {
+        aggregationIndex = newAggregations.length;
+      }
```

Comment:
Were we just not adding aggregations in this case before?

Reply:
This case never happened before, because the aggregation always existed and we just replaced it. Now we're removing it if the aggregation is empty.

Reply:
There is a minor difference in behaviour, where if you add a few aggregations, then edit the top one in the list, deselect everything, and then start selecting stuff again, it will now be at the end of the list.
I'm just going to change it to remove it only after the user exits the AggregationEdit screen.